core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanOps.scala

```diff
@@ -47,9 +49,38 @@ trait SpanOps[F[_]] {
     * }}}
     *
     * @see
-    *   [[use]], [[use_]], or [[surround]] for a managed lifecycle
+    *   [[use]], [[use_]], [[surround]], or [[resource]] for a managed lifecycle
     */
-  def startUnmanaged(implicit ev: Result =:= Span[F]): F[Span[F]]
+  def startUnmanaged: F[Span[F]]
+
+  /** Creates a [[Span]] and a [[cats.effect.Resource Resource]] for using it.
+    * Unlike [[startUnmanaged]], the lifecycle of the span is fully managed.
+    *
+    * The finalization strategy is determined by [[SpanFinalizer.Strategy]]. By
+    * default, the abnormal termination (error, cancelation) is recorded.
+    *
+    * If the start timestamp is not configured explicitly in a builder, the
+    * `Clock[F].realTime` is used to determine the timestamp.
+    *
+    * `Clock[F].realTime` is always used as the end timestamp.
+    *
+    * @see
+    *   default finalization strategy [[SpanFinalizer.Strategy.reportAbnormal]]
+    * @example
+    *   {{{
+    * val tracer: Tracer[F] = ???
+    * val ok: F[Unit] =
+    *   tracer.spanBuilder("resource-span")
+    *     .build
+    *     .resource
+    *     .use { case (span, wrap) =>
+    *       // `wrap` encloses its contents within the "resource-span" span;
+    *       // anything not applied to `wrap` will not end up in the span
+    *       wrap(span.setStatus(Status.Ok, "all good"))
+    *     }
+    *   }}}
+    */
+  def resource: Resource[F, (Span[F], F ~> F)]
```

Comment:
Would it make sense to use `SpanOps.Res[F]` instead of a tuple?

For example:
```scala
object SpanOps {
  sealed abstract class Res[F[_]] {
    def span: Span[F]
    def wrap: F ~> F
  }
}
```

I see a few pros:
1) Easier to deal with bincompat issues in the future
2) Fields can be adequately documented
3) Maybe a taste preference, but it could be more friendly for users

Wdyt?

Reply:
I like the idea, though if we're going to formalise it in a type, I think some bikeshedding on the name `wrap` might be in order. perhaps `include`?

I'll work on a commit, and we can bikeshed ad nauseam after

Reply:
I don't have a strong opinion regarding the name. `trace` could fit too.

Reply:
after quite a bit of thought, I have decided that `trace` is a better name than `include`